### PR TITLE
HBASE-22338 LICENSE file only contains Apache 2.0. 

### PR DESCRIPTION
**Pull request by rabi-kumar**
Added license and notice files in each of the sub-module.

### CODE REVIEW

File: kafka/hbase-kafka-proxy/src/main/appended-resources/META-INF/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,3610 @@
...
+This product includes jersey-client, jersey-core, jersey-json, jersey-server, jersey-guice,
+JAXB RI, javax.annotation API, Java Servlet API, JavaServer Pages(TM) API, javax.ws.rs-api,
+JAXB API bundle for GlassFish V3, Expression Language 3.0, HK2 API module, ServiceLocator,
+HK2 Implementation Utilities, OSGi resource locator bundle - used by various API
+providers that rely on META-INF/services mechanism to locate providers, aopalliance version
+1.0 repackaged as a module, javax.inject:1 as OSGi bundle,JSP implementation,
+jersey-repackaged-guava, jersey-container-servlet-core, jersey-core-client,
+jersey-core-common, jersey-core-server, jersey-media-jaxb, under the
+license COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.1
+The GNU General Public License (GPL) Version 2 with CLASSPATH EXCEPTION
```
Comment (busbey) [inline kafka/hbase-kafka-proxy/src/main/appended-resources/META-INF/LICENSE.txt]: Why is this here? are the above things dual licensed? I think so? If so, we should just say "under CDDL v1.1" and leave out the GPL stuff. we choose which license we redistribute under, and as an ASF project we can't choose GPL.

File: kafka/hbase-kafka-proxy/src/main/appended-resources/META-INF/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,3610 @@
...
+or (ii)�the combination of the Original Software with other software or devices.
+
+2.2. Contributor Grant.
+Conditioned upon Your compliance with Section 3.1 below and subject to
+third party intellectual property claims, each Contributor hereby grants
+You a world-wide, royalty-free, non-exclusive license:
+(a) under intellectual property rights (other than patent or trademark)
+ Licensable by Contributor to use, reproduce, modify, display, perform,
+ sublicense and distribute the Modifications created by such Contributor
+ (or portions thereof), eith
```
Comment (busbey) [inline kafka/hbase-kafka-proxy/src/main/appended-resources/META-INF/LICENSE.txt]: similarly to earlier, here we should say we're choosing the ALv2 license. as an asf project we can't distribute under LGPL.

File: spark/hbase-spark-it/src/main/appended-resources/META-INF/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,4498 @@
...
+INC. OR ANY CONTRIBUTORS TO THIS SOFTWARE BE LIABLE FOR ANY DIRECT, INDIRECT,
+INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+----
+This product includes Paranamer Core under the BSD License
```
Comment (busbey) [inline spark/hbase-spark-it/src/main/appended-resources/META-INF/LICENSE.txt]: we should try to aggregate these BSD entries (and repeated license text generally). instead just do something like

```
we include the following under Foo License:

* example 1, Copyright (c) XXX-XxXX foo bar catz
* example 2, Copyright (c) XXX- XXXX dee

Foo LICENSE

full text just once.
```

**Comment by ravowlga123** [comment]
Thank you for the detailed explanation @busbey . If there are issues and changes required then please do let me know.

**Comment by busbey** [comment]
This conversation on which LICENSE/NOTICE files need to change is still an issue AFAICT

https://github.com/apache/hbase-connectors/pull/41#discussion_r323340269

I did a quick glance at our pom files and I don't think we're including these dependencies in any of our generated jar files.


So I think the only place that should have these changes is the connectors assembly module since it controls what goes into our convenience tarball.

**Comment by ravowlga123** [comment]
@busbey If that is the case then I can aggregate these and add them in assembly module but in hbase-connectors-assembly aggregated notice and license files are auto-generated in target/META-INF/maven-shared-archive-resources. So I don't know if it's necessary to add them but please do let me know if it's needed.

**Comment by busbey** [comment]
yes I believe the assembly module is the only place we need to have this information.

the stuff that's getting autogenerated right now is the implementation based on the main repo that I mentioned earlier. you can turn it off by removing the plugins that drive the executions `aggregate-licenses` and `concat-NOTICE-files`.

**Comment by ravowlga123** [comment]
@busbey I have added the aggregated license and notice files in assembly module under src/main/resources and removed the plugins from assembly module which generated aggregated license and notice files for hbase-repo. Please do have a look at them in case if any changes are to be made.

**Comment by meszibalu** [comment]
@busbey @ravowlga123 do we need something for this change or can I merge it?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m  5s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  6s |  master passed  |
| -0 :warning: |  patch  |   0m 10s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  6s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  0s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  6s |  hbase-connectors-assembly in the patch passed.  |
|  |   |   4m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/41 |
| Optional Tests | dupname javac javadoc unit xml compile shellcheck shelldocs |
| uname | Linux 402af93e39fb 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 4c46a24 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| whitespace | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/whitespace-tabs.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/testReport/ |
| Max. process+thread count | 42 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly U: hbase-connectors-assembly |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/console |
| versions | git=2.20.1 shellcheck=0.5.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 52s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 11s |  master passed  |
| +1 :green_heart: |  compile  |   0m  8s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  8s |  master passed  |
| -0 :warning: |  patch  |   0m 11s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  6s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  7s |  hbase-connectors-assembly in the patch passed.  |
|  |   |   5m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/41 |
| Optional Tests | dupname javac javadoc unit xml compile shellcheck shelldocs |
| uname | Linux af1e070018d9 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 9a839b2 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| whitespace | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/whitespace-tabs.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/testReport/ |
| Max. process+thread count | 42 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly U: hbase-connectors-assembly |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/console |
| versions | git=2.20.1 shellcheck=0.5.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by taklwu** [comment]
will this block 1.0.1 release ? if so, I (or @rabi-kumar if he is still around) may pick this up.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 57s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  6s |  master passed  |
| -0 :warning: |  patch  |   0m 11s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  8s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  8s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  7s |  hbase-connectors-assembly in the patch passed.  |
|  |   |   4m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/41 |
| Optional Tests | dupname javac javadoc unit xml compile shellcheck shelldocs |
| uname | Linux 9ec8bbcc752c 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| whitespace | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/whitespace-tabs.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/testReport/ |
| Max. process+thread count | 46 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly U: hbase-connectors-assembly |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-Connectors-PreCommit/job/PR-41/1/console |
| versions | git=2.20.1 shellcheck=0.5.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 21s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 50s |  master passed  |
| +1 :green_heart: |  compile  |   0m  9s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  9s |  master passed  |
| -0 :warning: |  patch  |   0m 14s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  7s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  7s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  9s |  hbase-connectors-assembly in the patch passed.  |
|  |   |   9m 17s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/41 |
| Optional Tests | dupname javac javadoc unit xml compile shellcheck shelldocs |
| uname | Linux da9a557af2a7 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| whitespace | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-41/1/artifact/yetus-precommit-check/output/whitespace-tabs.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-41/1/testReport/ |
| Max. process+thread count | 43 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly U: hbase-connectors-assembly |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-41/1/console |
| versions | git=2.20.1 shellcheck=0.5.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by taklwu** [comment]
> yes I believe the assembly module is the only place we need to have this information.
> 
> the stuff that's getting autogenerated right now is the implementation based on the main repo that I mentioned earlier. you can turn it off by removing the plugins that drive the executions `aggregate-licenses` and `concat-NOTICE-files`.

sorry if above conversation is a long long time ago, I want to clarify your comment above,

if I read all your conversations correctly, @busbey you're basically saying we don't need this PR because below block of `hbase-connectors-assembly` has already been autogenerating the right `LICENSES` and `NOTICE` when we building the tarball. So can we close this PR and HBASE-22338? 


https://github.com/apache/hbase-connectors/blob/2bfc5f1c078e7530ca6506a397e74fa38668b1a5/hbase-connectors-assembly/pom.xml#L93-L126

I just open another one that fix the `NOTICE.aggregate`  #93

**Comment by taklwu** [comment]
https://github.com/apache/hbase-connectors/commit/036729dea0d1041e957f0349254f4ea1cf7cfba9 pushed, so closing this PR
